src/thor/bidirectional_astar.cc

```diff
@@ -1019,13 +1019,14 @@ std::vector<std::vector<PathInfo>> BidirectionalAStar::FormPath(GraphReader& gra
   while (edgelabel_index != kInvalidLabel) {
     const BDEdgeLabel& edgelabel = edgelabels_reverse_[edgelabel_index];
 
-    // Get elapsed time on the edge, then add the transition cost at
-    // prior edge.
+    // The first edge has no previous edge with elapsed cost so we can just use it directly
     uint32_t predidx = edgelabel.predecessor();
     if (predidx == kInvalidLabel) {
       cost += edgelabel.cost();
-    } else {
-      cost += edgelabel.cost() - edgelabels_reverse_[predidx].cost();
+    } // This edge needs the elapsed cost between it and the previous edge, since its the reverse path
+    // in flipping the path around we also need to switch out the transition costs (shifting right)
+    else {
+      cost += edgelabel.cost() - edgelabels_reverse_[predidx].cost() - edgelabel.transition_cost();
```

Comment:
@dnesbitt61 this seems like a long standing bug. though we were adding the correct turn cost below we forgot to also remove the incorrect turn cost from the adjacent edge. can you give this a look over to see if im crazy?

Reply:
@dgearhart @gknisely i suspect this will cause a bunch of RAD differences as well

Reply:
this bug was exposed on the 32bit build via the recosting unit tests btw. since the recosting  unit test only steps forward through the path it doesnt have to do gymnastics to copy the reverse path and mess with the costs like this.